Read generic network signal data hmi guidelines (#168)
smartdevicelink/sdl_hmi_integration_guidelines: docs/VehicleInfo/GetVehicleData/index.md

Purpose
: Get current values of specified vehicle data types.

!!! must

> 1. Consider a parameter in GetVehicleData request as `name` in case it is listed in HMI_API.xml; `name` is always boolean (according to API).  

Reviewer: ```suggestion
1. Map each parameter in GetVehicleData request to its `<name>` in the case that it is listed in HMI_API.xml; `<name>` is always a single boolean value (according to API).  
```

> 2. Consider a parameter in GetVehicleData request as `key` in case it is not listed in HMI_API.xml; `key` in GetVehicleData request may be either boolean or struct. Struct may contain either booleans or structs. 

Reviewer: ```suggestion
2. Map each parameter in GetVehicleData request to its `<key>` in the case that it is not listed in HMI_API.xml; `<key>` in GetVehicleData request may be either a boolean or struct value. If the value is a struct, it may contain either boolean or nested struct values, structured according to the vehicle data structure defined in `schema_items`. 
```

> 3. Respond with the correct schema_items to GetVehicleData request from SDL. 

Reviewer: ```suggestion
3. Respond with any custom vehicle data defined in `schema_items` included in the original request. 
```

> Keys and names should not match at different levels of nesting.   

Reviewer: should be a sub-bullet, rather than standalone text

Author: Fixed 5c09b11

Schema_items should not have duplicate `name` and `key` for different vehicle data items/sub-params, except same items is defined for different versions.  
Schema_items should not have `name`/`key` equivalent to any RPC vehicle data item/sub-param.

> 4. Respond with the structure as defined in API .xml

Reviewer: ```suggestion
4. Respond with any vehicle data defined in `HMI_API.xml` included in the original request.
```

GetVehicleData
![GetVehicleData](./assets/GetVehicleData.jpg)
|||
|||
GetVehicleData
![GetVehicleData](./assets/VDdefinedInXMLandSchema.png)

> |||

Reviewer: Missing  `|||`
```suggestion
|||
|||
```

> CustomVehicleData

Reviewer: ```suggestion
GetVehicleData with custom data
```
